### PR TITLE
Using lxml and Python for web scraping

### PR DESCRIPTION
**Pull request by yasoob**
This is the first post in a two post series on web scraping using lxml in Python. In this post I created the lxml based script. In the next post I will discuss how to convert this script into a Flask based API and how to host it on Heroku.

### CODE REVIEW

File: posts/web-scraping-using-lxml-python.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+Here is an example of a typical API response in JSON. This is the response from Reddit.
+
+![Typical API Response in JSON](https://imgur.com/WOBUbyn.png)
+
+There are a lot of Python libraries out there which can help you with web scraping. There is [lxml](http://lxml.de/), [BeautifulSoup](https://www.crummy.com/software/BeautifulSoup/?) and a full-fledged framework called [Scrapy](https://scrapy.org/). Most of the tutorials discuss lxml and Scrapy so I decided to go with lxml in this post. I will teach you the basics of XPaths and how you can use them to extract data from an HTML document. I will take you through a couple of different examples so that you can quickly get up-to-speed with lxml and XPaths.
```
Comment (sidhanthp) [inline posts/web-scraping-using-lxml-python.md]: I believe you said most of the tutorials talk about BeautifulSoup, not lxml and Scrapy.

File: posts/web-scraping-using-lxml-python.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+Here is an example of a typical API response in JSON. This is the response from Reddit:
+
+![Typical API Response in JSON](https://imgur.com/WOBUbyn.png)
+
+There are a lot of Python libraries out there which can help you with web scraping. There is [lxml](http://lxml.de/), [BeautifulSoup](https://www.crummy.com/software/BeautifulSoup/?) and a full-fledged framework called [Scrapy](https://scrapy.org/). Most of the tutorials discuss lxml and Scrapy, so I decided to go with lxml in this post. I will teach you the basics of XPaths and how you can use them to extract data from an HTML document. I will take you through a couple of different examples so that you can quickly get up-to-speed with lxml and XPaths.
```
Comment (sidhanthp) [inline posts/web-scraping-using-lxml-python.md]: I believe you said 'Most of the tutorials discuss BeautifulSoup and Scrapy'

**Review by sidhanthp — CHANGES_REQUESTED** [review]
Seriously a very well written article, we really appreciate the great job! I made a couple grammatical changes and have a couple quick changes for you to make.
1. Use headers to split the sections to make it easier for a person skimming the article to read it
2. For the screenshots of your terminal, box the part you are trying to attract the reader's attention to in red.

Seriously, fantastic job with the post!

File: posts/web-scraping-using-lxml-python.md (hunk 2 above)
Comment (yasoob) [reply posts/web-scraping-using-lxml-python.md]: I guess I must have been dozing while writing it. Thanks! I will fix it

**Comment by yasoob** [comment]
1. I will do that

2. As for the terminal, I am attracting the attention to the whole screenshot. It is just to let the readers know what the expected output is going to be once they run the listed commands. I think adding a red box is just going to make it messy when it doesn't need to be. Let me know your thoughts.

**Comment by sidhanthp** [comment]
Great, I found it a little difficult to follow the changes in the terminal (since much of the information was repeated). If you have any way to make the new information stand out, I think that would be a lot easier.

**Comment by yasoob** [comment]
I have added the headers in my last commit. I still don't think there is any sane way to add boxes in the terminal screenshots. If I were to draw them I would have to draw them around the whole terminal and that beats the purpose of the attention grabbing boxes.  😕 

Where exactly did you find yourself getting lost? All of the commands in the terminal are the exact same commands I mention before each screenshot.

**Comment by sidhanthp** [comment]
Not a big deal, the post looks good to go.

**Pull request merged**